system-addon/common/Reducers.jsm

```diff
@@ -60,7 +67,27 @@ function TopSites(prevState = INITIAL_STATE.TopSites, action) {
   }
 }
 
+function Search(prevState = INITIAL_STATE.Search, action) {
+  switch (action.type) {
+    case at.SEARCH_STATE_UPDATED:
+      if (!action.data) {
+        return prevState;
+      }
+      return Object.assign({}, prevState, {
+        currentEngine: {
+          name: action.data.currentEngine.name,
+          icon: action.data.currentEngine.iconBuffer
+        },
+        engines: action.data.engines.map(engine => ({
+          name: engine.name,
+          icon: engine.iconBuffer
```

Comment:
Just so i understand: construct the engines and current engines object in the search feed to look how we want it to look and pass it down to the reducer so all the reduce has to do is:
```js
  return Object.assign({}, prevState, {
        currentEngine: action.data.currentEngine,
        engines: action.data.engines
      });
``` 
Right?